sim_pipeline/image_simulation.py

```diff
@@ -144,3 +149,177 @@ def rgb_image_from_image_list(image_list, stretch):
+        [
+            (lens_pix_coordinate[0], lens_pix_coordinate[1]),
+            image_pix_coordinate,
+            ra_image_values[0],
+            dec_image_values[0],
+            image_amplitude[0],
+            image_magnitude[0],
+            np.array([ra_at_xy_0, dec_at_xy_0]),
+        ],
+        names=(
+            "deflector_pix",
+            "image_pix",
+            "ra_image",
+            "dec_image",
+            "image_amplitude",
+            "image_magnitude",
+            "radec_at_xy_0",
+        ),
+    )
+    return data
+
+
+def point_source_image(
+    lens_class,
+    band,
+    mag_zero_point,
+    delta_pix,
+    num_pix,
+    psf_kernels,
+    variability=None,
+    lensed=True,
+):
+    """Creates lensed point source images on the basis of given information.
+
+    :param lens_class: GalaxyGalaxyLens() object
+    :param band: imaging band
+    :param mag_zero_point: magnitude zero point in band
+    :param delta_pix: pixel scale of image generated
+    :param num_pix: number of pixels per axis
+    :param psf_kernels: psf kernels extracted from the dp0 cutout images
+    :param variability: None or list of variability function and time. Eg: variability =
+        {'time': t, 'function': sinusoidal_variability}, where t is a observation time
+        and sinusoidal_variability is a source variability function. If None, creates
+        images without variability.
+    :param lensed: True or False. If True, returns magnitude of lensed images.In this
+        function, it should be True.
+    :return: astropy table of deflector and image coordinate in pixel unit and other
+        properties
+    """
+
+    image_data = point_source_image_properties(
+        lens_class=lens_class,
+        band=band,
+        mag_zero_point=mag_zero_point,
+        delta_pix=delta_pix,
+        num_pix=num_pix,
+    )
+    transform_matrix = np.array([[delta_pix, 0], [0, delta_pix]])
+    grid = PixelGrid(
+        nx=num_pix,
+        ny=num_pix,
+        transform_pix2angle=transform_matrix,
+        ra_at_xy_0=image_data["radec_at_xy_0"][0],
+        dec_at_xy_0=image_data["radec_at_xy_0"][1],
+    )
+
+    ra_image_values = image_data["ra_image"]
+    dec_image_values = image_data["dec_image"]
+    amp = image_data["image_amplitude"]
+    magnitude = lens_class.point_source_magnitude(band, lensed=lensed)
+    psf_class = []
+    for i in range(len(psf_kernels)):
+        psf_class.append(PSF(psf_type="PIXEL", kernel_point_source=psf_kernels[i]))
+    # point_source_images = []
+    if variability is None:
+        point_source_images = []
+        for i in range(len(psf_class)):
+            rendering_class = PointSourceRendering(
+                pixel_grid=grid, supersampling_factor=1, psf=psf_class[i]
+            )
+            point_source = rendering_class.point_source_rendering(
+                np.array([ra_image_values[i]]),
+                np.array([dec_image_values[i]]),
+                np.array([amp[i]]),
```

Comment:
the amplitudes can also vary when the magnitude zero point is varying (which is normal for different exposures, or at least when the magnitude zero point is made equal, then the noise properties (Poisson noise) is different. We need to keep an eye on that

Reply:
Thanks Simon!